src/radiant.c

```diff
@@ -2853,6 +2888,73 @@ int radiant_get_scalers(radiant_dev_t * bd, int sc0, int sc1, uint16_t * scalers
   return 2*nscalers != radiant_get_mem(bd, DEST_FPGA, RAD_REG_SCALER_SCAL_BASE + sc0*2, nscalers*2, (uint8_t*) scalers); 
 }
 
+//slightly higher level that is used for the waveform_t
+int radiant_get_delays(radiant_dev_t * bd,radiant_readout_delay_t * rd)
+{
+  if(bd->rad_dateversion_int<=230)
+  {
+    printf("wrong fpga version - delays (and fpga memory) not implemented!!!\n");
+    return -1;
+  }
+  uint8_t rf0_group_mask=0;
+  uint8_t rf1_group_mask=0;
+
+  uint32_t group_mask[4]={0x1ff,0xe00,0x1ff000,0xe00000};
+  //group 0 = 0x1ff = channels 0-11
+  //group 1 = 0xe00 = channels 12-14
+  //group 2 = 0x1ff000 = channels 15-20
+  //group 3 = 0xe00000 = channels 21-23
+
+  //directly set the delay setting to the struct and get a temp group mask
+  radiant_get_delay_settings(bd,&rd->rf0_delay,&rd->rf1_delay,&rf0_group_mask,&rf1_group_mask);
+
+  //translate group mask to channel mask
+  uint8_t i =0;
+
+  for(i=0;i<4;i++)
+  {
+    if(rf0_group_mask&(1<<i)) rd->rf0_delay_mask|=group_mask[i];
+    if(rf1_group_mask&(1<<i)) rd->rf1_delay_mask|=group_mask[i];
+  }
+
+  return 0;
+}
+
+//closer to hardware level of how delays and masks are appiled
+int radiant_get_delay_settings(radiant_dev_t * bd, uint8_t * rf0_delay, uint8_t * rf1_delay, uint8_t * rf0_delay_mask,uint8_t * rf1_delay_mask)
+{
+  if(bd->rad_dateversion_int<=230)
+  {
+    printf("wrong fpga version - delays(and fpga memory) not implemented!!!\n");
+    return -1;
+  }
+  uint32_t delays[4];
+  radiant_get_mem(bd, DEST_FPGA, RAD_REG_LAB_CTRL_RF0_DELAY, 4*4,(uint8_t*)&delays);//lets just pull them all in
+
+  *rf0_delay=(delays[0]&0x7f);
```

Comment:
probably best to check if non-null before setting (in case someone only wants a subset of delays without having to create a dummy variable)